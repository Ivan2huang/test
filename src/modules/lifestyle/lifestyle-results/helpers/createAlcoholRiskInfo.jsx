import React from 'react';

import { AlcoholIcon, NoAlcoholIcon } from '../../../../icons';
import { formatMessage } from '../../../../helpers/helpers';

const createAlcoholRiskInfo = (intl, theme, alcoholRisk) => {
  const alcoholRiskMapping = {
    None: `${theme.lifeStyleResultOK}`,
    RegularDrinker: `${theme.lifeStyleResultWarning}`,
    HeavyDrinker: `${theme.lifeStyleResultAlarm}`,
  };

  return {
    id: 'alcohol',
    color: alcoholRiskMapping[alcoholRisk],
    CardIcon:
      alcoholRisk === 'None' ? (
        <NoAlcoholIcon />
      ) : (
        <AlcoholIcon color={alcoholRiskMapping[alcoholRisk]} />
      ),
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.alcoholRiskCard.title',
      'Alcohol',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.alcoholRiskCard.subTitle.${alcoholRisk}`,
      alcoholRisk,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.alcoholRiskCard.description.${alcoholRisk}`,
      alcoholRisk,
    ),
  };
};

export default createAlcoholRiskInfo;
