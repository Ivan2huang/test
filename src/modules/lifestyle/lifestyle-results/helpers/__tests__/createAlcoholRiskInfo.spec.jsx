import createAlcoholRiskInfo from '../createAlcoholRiskInfo';

jest.mock('../../../../../helpers/helpers', () => ({
  formatMessage: jest.fn((intl, id, defaultMessage) => defaultMessage),
}));

describe('Create Alcohol Risk Info', () => {
  const intl = {};
  const theme = {
    lifeStyleResultWarning: 'orange',
    lifeStyleResultOK: 'green',
    lifeStyleResultAlarm: 'red',
  };

  it('should match snapshot', () => {
    const actual = createAlcoholRiskInfo(intl, theme, 'None');

    expect(actual).toMatchSnapshot();
  });

  it('should match snapshot for drinker', () => {
    const actual = createAlcoholRiskInfo(intl, theme, 'HeavyDrinker');

    expect(actual).toMatchSnapshot();
  });
});
