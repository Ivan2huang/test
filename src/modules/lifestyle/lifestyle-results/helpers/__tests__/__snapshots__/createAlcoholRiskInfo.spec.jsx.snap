// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Alcohol Risk Info should match snapshot 1`] = `
Object {
  "CardIcon": <NoAlcohol
    color="#00847F"
    height="1.125rem"
    width="1.125rem"
  />,
  "color": "green",
  "description": "None",
  "id": "alcohol",
  "subTitle": "None",
  "title": "Alcohol",
}
`;

exports[`Create Alcohol Risk Info should match snapshot for drinker 1`] = `
Object {
  "CardIcon": <Alcohol
    color="red"
    height="1.125rem"
    width="1.125rem"
  />,
  "color": "red",
  "description": "HeavyDrinker",
  "id": "alcohol",
  "subTitle": "HeavyDrinker",
  "title": "Alcohol",
}
`;
